packages/react/src/components/TextArea/TextArea.tsx

```diff
@@ -164,12 +170,26 @@ const TextArea = React.forwardRef((props: TextAreaProps, forwardRef) => {
   );
   const { current: textAreaInstanceId } = useRef(getInstanceId());
 
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
```

Comment:
Refactored this file a bit since it was quite scattered. I've basically moved all refs to the top, followed by effects, then class variable declarations, then normalized JSX elements. The only net new code is [L222](https://github.com/carbon-design-system/carbon/pull/15115/commits/995f6cce992242ef539ecbb9670f1304dfd4e392#diff-af10a0e0381c771b14de261caf250073efd2136b83cbbc136c18fcfb0f80aaccR222), [L320](https://github.com/carbon-design-system/carbon/pull/15115/commits/995f6cce992242ef539ecbb9670f1304dfd4e392#diff-af10a0e0381c771b14de261caf250073efd2136b83cbbc136c18fcfb0f80aaccR320-R327), and [L345](https://github.com/carbon-design-system/carbon/pull/15115/commits/995f6cce992242ef539ecbb9670f1304dfd4e392#diff-af10a0e0381c771b14de261caf250073efd2136b83cbbc136c18fcfb0f80aaccR345)